openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,554 @@
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        Bokeh figure
+        """
+        # Sets data for x/y inputs
+        for title in self.input_data_dict:
+            if title == self.x_input.value:
+                x_value = self.x_input_slider.value
+                # Rounds the x_data to match the input_data_dict value
+                mach_index = np.where(
+                    np.around(self.input_data_dict[title], 5) == np.around(x_value, 5))[0]
+            elif title == self.y_input.value:
+                y_data = self.input_data_dict[title]
+
+        # Make slice in Z data at the point calculated before and add it to the data source
+        z_data = self.Z[:, mach_index].flatten()
+        self.source.add(z_data, 'left_slice')
+
+        x = self.source.data['left_slice']
+        y = self.slider_source.data[self.y_input.value]
+
+        # Create and format figure
+        right_plot_fig = figure(plot_width=200, plot_height=500, x_range=(min(x), max(x)),
+                                y_range=(min(y_data), max(y_data)),
+                                title="{} vs {}".format(self.y_input.value,
+                                                        self.output_select.value),
+                                tools="")
+        right_plot_fig.xaxis.axis_label = self.output_select.value
+        right_plot_fig.yaxis.axis_label = self.y_input.value
+        right_plot_fig.line(x, y)
+
+        # Determine distance and alpha opacity of training points
+        data = self._training_points()
+        vert_color = np.zeros((len(data), 1))
+        for i, info in enumerate(data):
+            alpha = np.abs(info[0] - x_value) / self.limit_range[self.x_index]
+            if alpha < self.dist_range:
+                vert_color[i, -1] = (1 - alpha / self.dist_range) * info[-1]
+
+        color = np.column_stack((data[:, -4:-1] - 1, vert_color))
+        alphas = [0 if math.isnan(x) else x for x in color[:, 3]]
+        right_plot_fig.scatter(x=data[:, 3], y=data[:, 1], line_color=None, fill_color='#000000',
+                               fill_alpha=alphas)
+
+        # Set the right_plot data source to new values
+        self.right_plot_source.data = dict(
+            left_slice_x=np.repeat(x_value, self.resolution), left_slice_y=y_data,
+            x1=np.array([x + self.dist_range for x in np.repeat(x_value, self.resolution)]),
+            x2=np.array([x - self.dist_range for x in np.repeat(x_value, self.resolution)]))
+
+        self.contour_plot.line('left_slice_x', 'left_slice_y', source=self.right_plot_source,
+                               color='black', line_width=2)
+
+        return right_plot_fig
+
+    def _bot_plot(self):
+        """
+        Create the bottom subplot to view the projected slice.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        Bokeh figure
+        """
+        for title in self.input_data_dict:
```

Comment:
is the loop necessary here?  couldn't you just use `x_input.value` and  `y_input.value` directly? 
```
self.x_data = self.input_data_dict[self.x_input_value]
self.y_value = self.y_input_slider.value
alt_index = np.where(np.around(
                    self.input_data_dict[self.y_input_value], 5) == np.around(self.y_value, 5))[0]
```